### PR TITLE
Simplify UnsafeStaticsAnalyzer

### PR DESCRIPTION
**Pull request by j3parker**
There's still some more to go but this is getting pretty close. End goal is still to merge with `ImmutabilityAnalyzer` into one mega analyzer ;)

I tested this by turning the unnecessary annotations up to error level and building the LMS with these changes. The only diagnostics it emitted were due to the #319 bug.

### CODE REVIEW

File: src/D2L.CodeStyle.Analyzers/Immutability/UnsafeStaticsAnalyzer.cs [hunk 1]
```diff
@@ -143,44 +139,36 @@ MutabilityInspector inspector
...
-					location,
-					fieldOrPropertyName
+			bool hasAnnotation = hasUnauditedAnnotation || hasAuditedAnnotation;
+
+			if ( !isStatic && hasAnnotation ) {
+				// non-statics don't need these annotations.
+				// TODO: Consider dropping the static-specific annotations in
+				// favour of Mutability.Audited/Unaudited. Will require some
+				// effort due to the reporting we do but it'll make things
+				// simpler once we merge the analyzers.
```
Comment (j3parker) [inline src/D2L.CodeStyle.Analyzers/Immutability/UnsafeStaticsAnalyzer.cs]: Once you get the unnecessary diagnostics working for `Mutability.*` attributes I'll investigate this. The benefit would be simplifying the code (which is more obvious when considering how to merge the analyzers.)

File: src/D2L.CodeStyle.Analyzers/Immutability/UnsafeStaticsAnalyzer.cs [hunk 2]
```diff
@@ -216,50 +216,17 @@ string fieldOrPropertyName
...
-
-		/// <summary>
-		/// All logic relating to either emitting or not emitting diagnostics other
-		/// than the ones about unnecessary annotations belong in this function.
-		/// This allows InspectMember to implement the logic around the unnecessary
-		/// annotations diagnostic. Any time we bail early in AnalyzeField or
-		/// AnalyzeProperty we risk not emitting unnecessary annotation
-		/// diagnostics.
-		/// </summary>
-		private IEnumerable<Diagnostic> GatherDiagnostics(
```
Comment (mtjandra) [inline src/D2L.CodeStyle.Analyzers/Immutability/UnsafeStaticsAnalyzer.cs]: Nice, yeah I was wondering about this portion when I first looked at it.

**Comment by mtjandra** [comment]
> There's still some more to go but this is getting pretty close. End goal is still to merge with ImmutabilityAnalyzer into one mega analyzer 

What the benefit to having a mega analyzer?

> I tested this by turning the unnecessary annotations up to error level and building the LMS with these changes. 

How are you doing this? Creating nuget package and referencing it during build from `references.ps1` ?

File: src/D2L.CodeStyle.Analyzers/Immutability/UnsafeStaticsAnalyzer.cs (hunk 2 above)
Comment (j3parker) [reply src/D2L.CodeStyle.Analyzers/Immutability/UnsafeStaticsAnalyzer.cs]: It made sense at one point I think.

**Comment by j3parker** [comment]
> What's the benefit to having a mega analyzer?

Shared cache (performance) and de-duplicating some logic. The analyzer then becomes "put the stuff that for whatever reason needs to be immutable through the inspector".

It would be a stepping stone to inferring `[Immutable]` for unsealed types (like interfaces) if they are `private`/`internal` but that's a bigger piece of work that I don't think is likely any time soon (not a lot of benefit for us at this point.)

> How are you doing this? Creating nuget package and referencing it during build from references.ps1 ?

Nope I just copy the DLL into the right subfolder of `checkout\packages\D2L.CodeStyle.Analyzers\` and do a quicky build.

**Pull request merged**